### ISSUE TITLE
Difficulty Reproducing CALVIN Results and Unstable Validation Loss

### ISSUE THREAD

**Issue by Wang-Ji-Zhe** [roboterax/video-prediction-policy#17]
Hi,
First, thank you for your excellent work on this project and for making the code publicly available.

I have been trying to reproduce the results of your paper on the CALVIN benchmark. I am using the code from this repository without any modifications and following the provided instructions.
You mention that a validation loss of around 0.06 is required to achieve the best performance. However, I'm facing some difficulties in reaching that target and replicating the reported evaluation scores.

Here's a summary of my observations after several training runs:
1. Validation Loss: My best training runs achieve a validation loss that hovers around 0.063-0.065, but it struggles to consistently drop to the 0.06 target.
2. Evaluation Performance: Consequently, my best evaluation on CALVIN results in an average successful task length of approximately 4.2, which is slightly below the paper's reported score of over 4.3.
3. Loss Curves: A peculiar behavior I've noticed is that the validation loss fluctuates significantly from the very beginning of the training process, even while the training loss shows a steady downward trend.

I have attached two plots from my best training run for your reference: the train/loss and val/loss curves.

Training Loss
<img width="544" height="549" alt="Image" src="https://github.com/user-attachments/assets/04906bb9-36db-472b-94a0-2215c29e837b" />

Validation Loss
<img width="544" height="375" alt="Image" src="https://github.com/user-attachments/assets/a5355c6b-74b0-4ed8-8941-1e1705bc2c37" />

I would be very grateful if you could provide some guidance on the following questions:
1. Is this volatile behavior for the validation loss expected?
2. Given that my validation loss is close to the target, could there be another factor I'm missing that is causing the performance gap on the CALVIN evaluation?
3. Do you have any suggestions on what I could check to better stabilize the validation loss and reproduce the reported results?
Thank you for your time and any help you can provide.

**Comment by Hu-Yuch** [comment]
Thank you for your interest in our work!
Your loss curve looks essentially correct—the final eval_loss typically fluctuates between 0.06–0.065. In the CALVIN benchmark, Diffusion Policy appears to be somewhat sensitive to checkpoint selection, so eval_loss only provides a rough indication of performance and does not directly correspond to the average rollout length.
In our experiments, we selected a later checkpoint (around 60k steps), which reached a score of 4.3. You may want to try evaluating multiple checkpoints as well. Also, you can try to increase batch size. In practice, the difference between 4.2 and 4.3 in CALVIN is negligible in terms of single-trial success rate.
Hope this helps!
